Demo/Sources/Recycling/GridViews/Markets/MarketsGridViewDemoView.swift

```diff
@@ -4,47 +4,229 @@
+        demoStack.heightAnchor.constraint(equalToConstant:450).isActive = true
+    }
 }
 
-class MarketsGridViewDemoView: UIView, Demoable {
-    lazy var dataSource: MarketDataSource = {
-        return MarketDataSource()
-    }()
+class FinnDataSource: NSObject {
+    var models = FinnMarket.newMarkets
+}
 
+class FinnMarketsDemoView: UIView, Demoable, MarketsViewDataSource, MarketsViewDelegate {
+    lazy var dataSource: FinnDataSource = {
+        return FinnDataSource()
+    }()
+    var titleLabel = UILabel()
     override init(frame: CGRect) {
         super.init(frame: frame)
-
         setup()
     }
-
     required init?(coder aDecoder: NSCoder) { fatalError() }
 
     private func setup() {
-        let collectionView = MarketsGridView(accessibilityHeader: "Markeder", delegate: self, dataSource: self)
+        titleLabel.text = "Finn"
+        titleLabel.font = UIFont.boldSystemFont(ofSize: 16.0)
+        titleLabel.textAlignment = .center
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(titleLabel)
+
+        var collectionView = MarketsGridView(accessibilityHeader: "Markeder", delegate: self, dataSource: self)
+        
         collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.isMarketGridCellLabelTwoLined = false
+        
         addSubview(collectionView)
+        
         NSLayoutConstraint.activate([
-            collectionView.topAnchor.constraint(equalTo: topAnchor),
+            titleLabel.topAnchor.constraint(equalTo: topAnchor, constant: 10),
+            titleLabel.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 20),
+
+            collectionView.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: 10),
             collectionView.leadingAnchor.constraint(equalTo: leadingAnchor),
             collectionView.trailingAnchor.constraint(equalTo: trailingAnchor),
-            collectionView.heightAnchor.constraint(equalToConstant: collectionView.calculateSize(constrainedTo: frame.width).height)
+            collectionView.heightAnchor.constraint(equalToConstant: 166)
         ])
     }
-}
 
-extension MarketsGridViewDemoView: MarketsViewDataSource {
-    func numberOfItems(inMarketsView marketsView: MarketsView) -> Int {
+    func numberOfItems(inMarketsView marketsView: FinniversKit.MarketsView) -> Int {
+        
         return dataSource.models.count
+
+    }
+    func marketsView(_ marketsView: FinniversKit.MarketsView, modelAtIndex index: Int) -> FinniversKit.MarketsViewModel {
+        return dataSource.models[index]
+
     }
+    func marketsView(_ marketsGridView: FinniversKit.MarketsView, didSelectItemAtIndex index: Int) {
+    }
+}
 
-    func marketsView(_ marketsView: MarketsView, modelAtIndex index: Int) -> MarketsViewModel {
+class FinnMarketsDemoView2: UIView, Demoable, MarketsViewDataSource, MarketsViewDelegate {
```

Comment:
I don't follow what's happening here. Could you please explain `FinnMarketsDemoView2`, especially that down in the setup the title label says `Tori` ? 🙂

Reply:
it was because I was testing before I made it to the toridemo and forgot to replace it